src/games/strategy/triplea/ai/proAI/data/ProPurchaseOptionMap.java

```diff
@@ -45,8 +47,10 @@ public ProPurchaseOptionMap(final PlayerID player, final GameData data) {
     specialOptions = new ArrayList<ProPurchaseOption>();
 
     // Add each production rule to appropriate list(s)
-    final List<ProductionRule> rules = player.getProductionFrontier().getRules();
-    for (final ProductionRule rule : rules) {
+    final ProductionFrontier productionFrontier = player.getProductionFrontier();
```

Comment:
So in this case this is a constructor and all the lists need to be initialized to avoid nulls.

I think the logging is fine as essentially that means there are no purchase options.

It is a long method that should eventually be refactored but out of scope here as this is just a small bug fix.